build/Makefile

```diff
@@ -501,7 +505,11 @@ testrun : test
 TEST_STATUS:
 	@echo "========================= Performing Building provider tests"
 
-test : TEST_STATUS $(SCXPAL_INTERMEDIATE_DIR) $(INTERMEDIATE_DIR)/testrunner
+TEST_DATA_PREP:
+	@echo "========================= Performing Test data preparation"
+	cp -R $(TEST_DIR)/omi_preexec/testdata/cnfs $(INTERMEDIATE_DIR)/.
```

Comment:
Yeah, that's what I thought. I think that, in the test, you should pass the prefix that should be added to all "open" files. So, for example, set up the tests like:

test/code/omi_preexec/testdata/cnfs/rhel/etc/mysql/mysql.cnf

You would pass in the full path to test/code/omi_preexec/testdata/cnfs/rhel as a prefix in the test object. The test object would prepend that to any !include or !includedir. In this way, you could check in REAL (unmodified) files from vendors and have them work perfectly, assuming your code is correct. I'm not wild changing the vendor files to work under test unless we have to and, in this case, we can make them work properly.

---
In reply to: [67187563](https://github.com/Microsoft/MySQL-Provider/pull/5#discussion_r67187563) [](ancestors = 67187563)